From f7ce0656d19724a78d5f9ff0aba095e1d28c3cbd Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Tue, 26 Mar 2013 16:55:38 +0800
Subject: [PATCH 434/635] ENGR00255796 FBIOPAN_DISPLAY error in earlysuspend
 state

SurfaceFlinger will use software vsync to feed those UI request
in earlysuspend state. Otherwise Activity will be blocked on requesting
UI buffer. In this case, PAN_DISPLAY failure should not be token as error.
* Make the error as warning
* Remove the duplicated unlock function

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx6/libgralloc_wrapper/framebuffer.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/mx6/libgralloc_wrapper/framebuffer.cpp b/mx6/libgralloc_wrapper/framebuffer.cpp
index f02abf5..e7ecad8 100755
--- a/mx6/libgralloc_wrapper/framebuffer.cpp
+++ b/mx6/libgralloc_wrapper/framebuffer.cpp
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2008 The Android Open Source Project
- * Copyright (C) 2010-2012 Freescale Semiconductor, Inc.
+ * Copyright (C) 2010-2013 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -127,8 +127,7 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
         m->info.yoffset = offset / m->finfo.line_length;
 
         if (ioctl(m->framebuffer->fd, FBIOPAN_DISPLAY, &m->info) == -1) {
-            ALOGE("<%s, %d> ioctl FBIOPAN_DISPLAY failed", __FUNCTION__, __LINE__);
-            m->base.unlock(&m->base, buffer); 
+            ALOGW("FBIOPAN_DISPLAY failed: %s", strerror(errno));
             m->currentBuffer = buffer;
             return 0;
             //return -errno;
-- 
1.8.0

